code/modules/food_and_drinks/food/snacks_pie.dm

```diff
@@ -26,24 +26,28 @@
 	tastes = list("pie" = 1)
 
 /obj/item/weapon/reagent_containers/food/snacks/pie/cream/throw_impact(atom/hit_atom)
-	if(!..()) //was it caught by a mob?
-		var/turf/T = get_turf(hit_atom)
-		new/obj/effect/decal/cleanable/pie_smudge(T)
-		reagents.reaction(hit_atom, TOUCH)
-
-		if(ishuman(hit_atom))
-			var/mob/living/carbon/human/H = hit_atom
-			var/mutable_appearance/creamoverlay = mutable_appearance('icons/effects/creampie.dmi')
-			if(H.dna.species.id == "lizard")
-				creamoverlay.icon_state = "creampie_lizard"
-			else
-				creamoverlay.icon_state = "creampie_human"
-			H.Knockdown(20) //splat!
-			H.adjust_blurriness(1)
-			visible_message("<span class='userdanger'>[H] was creamed by [src]!!</span>")
-			H.add_overlay(creamoverlay)
-
-		qdel(src)
+	. = ..()
+	addtimer(CALLBACK(src, .proc/splat, hit_atom), 0.5)
```

Comment:
See, that's what I thought, but it doesn't work without the delay. I believe it's because I refactored catching to use `attack_hand`, so the impact (and subsequent deletion) occur _before_ the catch calls.